### PR TITLE
fix: multiple a11y fixes

### PR DESCRIPTION
**Pull request by travis1111**
Contributes to #

## What did you do?
Fix various accessibillty errors.

## Why did you do it?
Fix IBM accessibility checker failures.

## How have you tested it?

Tested on my local m2mui build.

## Were docs updated if needed?

- [ ] N/A
- [ x] No
- [ ] Yes

------------------------
`cv-checkbox.vue`:
Before:

image

image

`cv-codesnippet...`:

Before:

image

image

image

`cv-file-uploader.vue`:

Before:

image

image

`cv-modal.vue`:

Before:
image

image

`cv-multi-select.vue`:

Before

image

image

`cv-number-input.vue`:

Before:

image

image

`cv-search.vue`:

Before:
image

image

`cv-slider.vue`:

image

image

`cv-tabs.vue`:
Before:

image

image

`cv-toggle.vue`:
image

image

------------------------

### CODE REVIEW

File: packages/core/src/components/cv-search/cv-search.vue [hunk 1]
```diff
@@ -114,6 +115,17 @@ export default {
...
   },
   computed: {
+    wrapperAttrs() {
+      const attrs = {};
+      if (this.ariaLabel) {
+        attrs['aria-label'] = this.ariaLabel;
+      } else if (this.label) {
+        attrs['aria-labelledby'] = this.uid;
+      } else {
+        attrs['aria-label'] = "Search box";
```
Comment (lee-chase) [inline packages/core/src/components/cv-search/cv-search.vue:125]: Hard coded string

File: packages/core/src/components/cv-tabs/cv-tabs.vue [hunk 2]
```diff
@@ -43,16 +43,15 @@
               [`${carbonPrefix}--tabs--scrollable__nav-item--selected`]: selectedId == tab.uid,
             },
           ]"
-          role="presentation"
+          role="tab"
```
Comment (lee-chase) [inline packages/core/src/components/cv-tabs/cv-tabs.vue:46]: Carbon does not do this. Another PR removes 47 and 48.

**Review by lee-chase — CHANGES_REQUESTED** [review]
Hi @travis1111 

Mostly looks good, and many thanks for your work.

There are a couple of hard coded strings and some changes that Carbon React does not adopt in the Tabs. Where possible we should try to stick to core Carbon solutions.

File: packages/core/src/components/cv-search/cv-search.vue (hunk 1 above)
Comment (travis1111) [reply packages/core/src/components/cv-search/cv-search.vue:125]: That provides a default "arial-label" for the search box if the user did not pass in arialLable. I am ok to remove that.  Please let me know if this is your intention (to not have adefault arial-labels value).  Thanks

**Comment by lee-chase** [comment]
@travis1111 

Notes/queries:
- CvCheckbox change 👍 
- CodeSnippet - do not understand why the aria-hidden has been replaced by a label, the textarea should be hidden. This seems to be combined with the addition of a new `ariaLabel` property which is not used.
- CvModal - why no aria-label for passive alert modal?
- CvMultiselect - can this not use labelled by?
- CvNumberInput - Carbon uses a `hideLabel` property for this visually hiding the label, but not removing it.
- CvSearch - Carbon visually hides the label and does not hide the DOM element.
- CvSlider -  Carbon visually hides the label and does not hide the DOM element.
- CvTabs - this is not consistent with Carbon. Will #1332 have resolved this for you?
- CvToggle change 👍 

Tempted to allow some of the label hiding but @carbon/vue is an implementation of Carbon based on @carbon/react as the canonical version. Can we mimic Carbon by adding `bx--visually-hidden` classes.

**Comment by travis1111** [comment]
> @travis1111
> 
> Notes/queries:
> 
> * CvCheckbox change 👍
> * CodeSnippet - do not understand why the aria-hidden has been replaced by a label, the textarea should be hidden. This seems to be combined with the addition of a new `ariaLabel` property which is not used.
> * CvModal - why no aria-label for passive alert modal?
> * CvMultiselect - can this not use labelled by?
> * CvNumberInput - Carbon uses a `hideLabel` property for this visually hiding the label, but not removing it.
> * CvSearch - Carbon visually hides the label and does not hide the DOM element.
> * CvSlider -  Carbon visually hides the label and does not hide the DOM element.
> * CvTabs - this is not consistent with Carbon. Will [fix: address a11y issue raised in #1332](https://github.com/carbon-design-system/carbon-components-vue/pull/1332) have resolved this for you?
> * CvToggle change 👍
> 
> Tempted to allow some of the label hiding but @carbon/vue is an implementation of Carbon based on @carbon/react as the canonical version. Can we mimic Carbon by adding `bx--visually-hidden` classes.

Thanks I will look into this shortly

**Comment by travis1111** [comment]
> CvNumberInput - Carbon uses a hideLabel property for this visually hiding the label, but not removing it.

I searched components, it uses `bx--visually-hidden` to hide it, but it does not work, `bx--visually-hidden` does not hide the component:

image

**Comment by travis1111** [comment]
@lee-chase Thanks for reviewing the code.

> CodeSnippet - do not understand why the aria-hidden has been replaced by a label, the textarea should be hidden. This seems to be combined with the addition of a new ariaLabel property which is not used.

Addressed

> CvModal - why no aria-label for passive alert modal?

Addressed

> CvMultiselect - can this not use labelled by?

Addressed

> CvNumberInput - Carbon uses a hideLabel property for this visually hiding the label, but not removing it.

Changed to `v-show`, please see comments above.

> CvSearch - Carbon visually hides the label and does not hide the DOM element.
> CvSlider - Carbon visually hides the label and does not hide the DOM element.

Same as number input, the hidden class is not working.  Is `visually hides ` refers to `visibility: hide`? 

> CvTabs - this is not consistent with Carbon. Will https://github.com/carbon-design-system/carbon-components-vue/pull/1332 have resolved this for you?

Yes, that PR fixes the problem, I have reverted my changes.

Could you please taker a look again?

Thanks

**Comment by lee-chase** [comment]
@travis1111 

- CvMultiselect - can this not use labelled by?
Not quite what I meant so I pushed an update. Fingers crossed the tests pass.

**Review by lee-chase — APPROVED** [review]
LGTM, assuming tests pass.

**Review by lee-chase — APPROVED** [review]
LGTM

**Comment by travis1111** [comment]
thanks @lee-chase

**Pull request merged**